src/components/date-picker/day.tsx

```diff
@@ -1,44 +1,42 @@
-import * as React from 'react';
-
 import {computed} from 'mobx';
 import {observer} from 'mobx-react';
-
+import * as React from 'react';
+import { SBComponent } from 'stylable-react-component';
+import {root} from 'wix-react-tools';
 import styles from './date-picker.st.css';
 
 export interface DayProps {
-    day: number | string;
+    day: number;
     selected?: boolean;
     currentDay?: boolean;
     focused?: boolean;
     partOfPrevMonth?: boolean;
     partOfNextMonth?: boolean;
-    dataAutomationId?: string;
-    onSelect?(day: number | string): void;
+    onSelect?(day: number): void;
 }
 
+@SBComponent(styles)
 @observer
 export class Day extends React.Component<DayProps, {}> {
-    @computed
-    get styles(): string {
-        if (this.props.focused) {
-            return [styles.calendarItem, styles.day, styles.selectedDay].join(' ');
-        } else if (this.props.selected) {
-            return [styles.calendarItem, styles.day, styles.selectedDay].join(' ');
-        } else if (this.props.currentDay) {
-            return [styles.calendarItem, styles.day, styles.currentDay].join(' ');
-        } else if (this.props.partOfNextMonth || this.props.partOfPrevMonth) {
-            return [styles.calendarItem].join(' ');
-        } else {
-            return [styles.calendarItem, styles.day].join(' ');
-        }
-    }
-
     public render() {
+        const cssStates = {
+            focused: this.props.focused!,
+            selected: this.props.selected!,
+            current: this.props.currentDay!,
+            inactive: this.props.partOfNextMonth! || this.props.partOfPrevMonth!,
```

Comment:
seems like `inactive` state can be derived from `active` (or vice versa). you can use `:not(:active)` selector in your st.css file